lib/src/main/java/com/auth0/android/lock/provider/AuthResolver.java

```diff
@@ -0,0 +1,52 @@
+package com.auth0.android.lock.provider;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.auth0.android.provider.AuthHandler;
+import com.auth0.android.provider.AuthProvider;
+
+import java.util.List;
+
+
+/**
+ * Holds instances of AuthHandlers that can be used to query for AuthProviders given
+ * a strategy and connection name.
+ * When no AuthProvider is matched, it will return null
+ */
+public abstract class AuthResolver {
```

Comment:
@nikolaseu Both options are good. Although using a private constructor has the advantage of not letting the user to extend the abstract class and instantiate objects. I will change it. 👍